src/logic/operate.js

```diff
@@ -0,0 +1,24 @@
+import Big from 'big.js';
+
+const operate = (numberOne, numberTwo, operation) => {
+  const display = document.querySelector('.displayOutput');
+  let result = '';
+  const firstNumber = new Big(numberOne);
+  const secondNumber = new Big(numberTwo);
+
+  if (operation === '+') {
+    result = firstNumber.plus(secondNumber);
+  } else if (operation === '-') {
+    result = firstNumber.minus(secondNumber);
+  } else if (operation === 'x') {
+    result = firstNumber.times(secondNumber);
+  } else if (operation === '/') {
+    result = firstNumber.div(secondNumber);
+  } else if (operation === '%') {
```

Comment:
- [ ] `%` in our case isn't `modulus`, it's just converting a single number to a percentage

![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/performOperate.png)